### DISCUSSION TITLE
How to perform an UPSERT operation?

### DISCUSSION THREAD

**Discussion by agersant** [vincent-herlemont/native_db#239] [Q&A] [answered]
Hi and thanks for writing `native_db`. I am evaluating it for the next major version of https://github.com/agersant/polaris and I am curious what is the recommended way of performing an UPSERT (update or insert) operation for a given primary key?

**Comment by vincent-herlemont** [comment, marked as answer]
You are correct, there was no upsert function on the existing release of Native DB. This has been added for version 0.8.0, which will be released next week. (See code [`upsert`](https://github.com/vincent-herlemont/native_db/blob/c29062ea9324b776514e23b2c16a81cd26a2f60d/src/transaction/rw_transaction.rs#L140-L196)).

If you want to perform an upsert with previous versions, you can check for existence using [`get`](https://docs.rs/native_db/latest/native_db/transaction/struct.RwTransaction.html#method.get) and use [`insert`](https://docs.rs/native_db/latest/native_db/transaction/struct.RwTransaction.html#method.insert) or [`update`](https://docs.rs/native_db/latest/native_db/transaction/struct.RwTransaction.html#method.update) depending on whether the data exists.
